### PR TITLE
Test-CXTB-255 Onboarders Create New Participant

### PR DESCRIPTION
**Pull request by cmoreno-nts**
-Create Test

image

### CODE REVIEW

File: tests/page_objects/care_platform_page.yaml [hunk 1]
```diff
@@ -95,6 +96,56 @@ care_platform_participant:
...
+  insurance_member_id_input: //input[@id= 'insuranceMemberIdentification']
+  insurance_group_id_input: //input[@id='insuranceGroupIdentification']
+  insurance_provider_address_input: //input[@id='insuranceAddress']
+  insurance_provider_address_numbers_input: //input[@id='insuranceAddress2']
+  insurance_provider_address_city_input: //input[@id='insuranceCity']
+  insurance_provider_address_state_dropdown_button: //select[@id='insuranceState']
+  insurance_provider_address_state_option: //select[@id='insuranceState']//option[contains(text(),'OH')]
+  insurance_provider_address_zip_input: //input[@id='insuranceZip']
+  race_checkbox: //span[contains(text(),'Asian')]
+  emergency_contact_first_name_input: //input[@id='emergencyContactFirstName']
```
Comment (jmontero-nts) [inline tests/page_objects/care_platform_page.yaml:128]: use id instead of xpath

File: tests/page_objects/care_platform_page.yaml [hunk 2]
```diff
@@ -95,6 +96,56 @@ care_platform_participant:
...
+  emergency_contact_last_name_input: //input[@id='emergencyContactLastName']
+  emergency_contact_phone_number_input: //input[@id='emergencyContactPhone']
+  emergency_contact_email_input: //input[@id='emergencyContactEmailAddress']
+  emergency_contact_relationship_dropdown_button: //select[@id='emergencyContactRelationship']
+  emergency_contact_relationship_options: //select[@id='emergencyContactRelationship']//option[contains(text(),'Other family member')]
+  emergency_contact_address_input: //input[@id='emergencyContactAddress']
```
Comment (jmontero-nts) [inline tests/page_objects/care_platform_page.yaml:134]: all xpath that only contains attributes id use Strategy: id in the related test.

**Review by jmontero-nts — CHANGES_REQUESTED** [review]

File: tests/page_objects/care_platform_page.yaml (hunk 1 above)
Comment (cmoreno-nts) [reply tests/page_objects/care_platform_page.yaml]: Suggested changes already applied.

File: tests/page_objects/care_platform_page.yaml (hunk 2 above)
Comment (cmoreno-nts) [reply tests/page_objects/care_platform_page.yaml]: Suggested changes already applied.

**Review by jmontero-nts — DISMISSED** [review]

**Review by miguel-hquez — CHANGES_REQUESTED** [review]

File: tests/page_objects/care_platform_page.yaml [hunk 3]
```diff
@@ -81,6 +81,7 @@ care_platform_logout_modal:
 
 care_platform_participant:
   participants_title: //div[@class='text-lg font-bold']//h3[@class='text-lg font-bold']
+  add_participant_button: //button[text()='Add'] #Add Button
```
[the remark this answers is not in the document]
Comment (cmoreno-nts) [reply tests/page_objects/care_platform_page.yaml:84]: it was for local proposes, I will remove it.

File: lib/core/device.rb [hunk 4]
```diff
@@ -1933,6 +1933,30 @@ def return_element_attribute(action)
 
 end
 
+# Returns a variable with a unique number adding a ramdon number at the end
+# i.e. method receives "Hey" and then returns "829225<fjh>"
+def generate_unique_number(action)
```
Comment (jmontero-nts) [inline lib/core/device.rb:1938]: It is only generate number 0 to 10. is it?

Comment (csanchez-nts) [inline lib/core/device.rb:1938]: You need to add this action to the readme file

File: lib/core/device.rb [hunk 5]
```diff
@@ -1933,6 +1933,30 @@ def return_element_attribute(action)
...
+  name = convert_value(action["Name"])
+  unique_number = "#{name}#{2.times.map{(0...(rand(10))).map { ('1'..'9').to_a[rand(26)] }.join }.join("")}"
+  ENV[convert_value(action["ResultVar"])] = unique_number
+end
+
+# Returns a variable with a unique email adding a ramdon string at the begining
+# i.e. method receives "@gmail.com" and then returns "<fjh>@gmail.com"
+def generate_unique_email(action)
```
Comment (csanchez-nts) [inline lib/core/device.rb:1946]: same here

File: lib/core/device.rb [hunk 6]
```diff
@@ -1933,6 +1933,30 @@ def return_element_attribute(action)
...
+  name = convert_value(action["Name"])
+  unique_email = "#{1.times.map{(0...(rand(10))).map { ('a'..'z').to_a[rand(26)] }.join }.join("")}#{name}"
+  ENV[convert_value(action["ResultVar"])] = unique_email
+end
+
+# Returns a variable with a unique name adding a ramdon string at the end
+# i.e. method receives "Hey" and then returns "Hey<fjh>"
+def generate_unique_string(action)
```
Comment (csanchez-nts) [inline lib/core/device.rb:1954]: and here

**Review by csanchez-nts — APPROVED** [review]

**Review by csanchez-nts — DISMISSED** [review]

**Review by jmontero-nts — DISMISSED** [review]

**Review by miguel-hquez — APPROVED** [review]

File: tests/page_objects/onboarders_care_platform_page.yaml [hunk 7]
```diff
@@ -1,3 +1,10 @@
+onboarders_platform_home:
+  page_title: //h3
```
Comment (jmontero-nts) [inline tests/page_objects/onboarders_care_platform_page.yaml:2]: Where is this used?

File: tests/page_objects/onboarders_care_platform_page.yaml [hunk 8]
```diff
@@ -1,3 +1,10 @@
...
+  call_queue_title: //h3[text()='Call Queue']
```
Comment (jmontero-nts) [inline tests/page_objects/onboarders_care_platform_page.yaml:3]: this one too, where is it used? if there are new page objects that are not used, delete them.

**Review by jmontero-nts — DISMISSED** [review]

File: README.md [hunk 9]
```diff
@@ -1183,6 +1183,33 @@ Prints and Writes now time + # minutes.
...
+    - Type: generate_unique_number
+      Name: number that user needs to be unique
+      ResultVar: Var that will have the unique number value
+
+### <a id="generate_unique_email"></a>generate_unique_email
+
+Returns a variable with a unique email using ramdon string at the begining 
+method receives "@gmail.com" and then returns "ujrb@gmail.com"
+
+    - Type: generate_unique_email
```
Comment (csanchez-nts) [inline README.md:1200]: This description is not needed anymore as the method was removed

**Review by csanchez-nts — CHANGES_REQUESTED** [review]

**Review by dguzmanneural — CHANGES_REQUESTED** [review]

File: tests/page_objects/onboarders_care_platform_page.yaml [hunk 10]
```diff
@@ -16,6 +23,73 @@ onboarders_notifications:
...
+  search_input: //input[@id='searchText']
+  clear_search_button: //button[contains(@class,'button-circle')]
+  status_dropdown_button: //button[contains(@class,'queue-filter-dropdown-toggle')]
+  status_dropdown_options: //div[contains(@class, 'queue-filter-dropdown')]  
+  participant_info: participant-component
+  records_button: //button[text()='Records']
+
+onboarders_participant_record:
+  info_right_column: //div[@class='flex-col flex-col-divider']//h4[text()='Community Support / Backup Plan']
+  participants_device_login: //*//h4[text()='Participants Device Login']
```
[the remark this answers is not in the document]
Comment (cmoreno-nts) [reply tests/page_objects/onboarders_care_platform_page.yaml:47]: Done

File: tests/page_objects/onboarders_care_platform_page.yaml [hunk 11]
```diff
@@ -16,6 +23,73 @@ onboarders_notifications:
...
+  records_button: //button[text()='Records']
+
+onboarders_participant_record:
+  info_right_column: //div[@class='flex-col flex-col-divider']//h4[text()='Community Support / Backup Plan']
+  participants_device_login: //*//h4[text()='Participants Device Login'] 
+  participants_username: //strong[contains(text(), 'Username')]/../following-sibling::div
+  participants_password: //strong[contains(text(), 'Password')]/../following-sibling::div
+
+onboarders_add_participant_form:
+  create_participant_title: //div[@class='flex flex-wrap flex-half-gutters']
```
[the remark this answers is not in the document]
Comment (cmoreno-nts) [reply tests/page_objects/onboarders_care_platform_page.yaml:52]: Done

File: tests/page_objects/onboarders_care_platform_page.yaml [hunk 12]
```diff
@@ -67,4 +141,4 @@ onboarder_edit_staff_information:
   first_name_error_label: //form//input[@id='firstName']/following-sibling::div[@aria-hidden='false']
   last_name_input: //form//input[@id='lastName']
   last_name_error_label: //form//input[@id='lastName']/following-sibling::div[@aria-hidden='false']
-  phone_input: //form//input[@id='phone']
\ No newline at end of file
+  phone_input: //form//input[@id='phone']
```
[the remark this answers is not in the document]
Comment (cmoreno-nts) [reply tests/page_objects/onboarders_care_platform_page.yaml:144]: It seem that was a mistake, I had no reason to do it.

[the remark this answers is not in the document]
Comment (cmoreno-nts) [reply tests/page_objects/onboarders_care_platform_page.yaml:144]: Its seem that is detecting as a deletion, but this is on the main branch.

File: tests/cases/helpers/case_helpers_care_platform_providers.yaml [hunk 13]
```diff
@@ -21,11 +21,11 @@ ProvidersCarePlatformLogin:
     - Type: send_keys
       Strategy: id
       Id: $PAGE.care_platform_login.email_input$
-      Value: $AND_CLI_provider_user1$
+      Value: $AND_CLI_provider_user2$
     - Type: send_keys
```
[the remark this answers is not in the document]
Comment (cmoreno-nts) [reply tests/cases/helpers/case_helpers_care_platform_providers.yaml]: The one for Onboarders was not created, anyways I added.

File: tests/page_objects/onboarders_care_platform_page.yaml [hunk 14]
```diff
@@ -16,6 +23,73 @@ onboarders_notifications:
...
+  search_button: //span[@class='icon-search']
+  search_input: //input[@id='searchText']
+  clear_search_button: //button[contains(@class,'button-circle')]
+  status_dropdown_button: //button[contains(@class,'queue-filter-dropdown-toggle')]
+  status_dropdown_options: //div[contains(@class, 'queue-filter-dropdown')]  
+  participant_info: participant-component
+  records_button: //button[text()='Records']
+
+onboarders_participant_record:
+  info_right_column: //div[@class='flex-col flex-col-divider']//h4[text()='Community Support / Backup Plan']
```
[the remark this answers is not in the document]
Comment (cmoreno-nts) [reply tests/page_objects/onboarders_care_platform_page.yaml]: Done

File: README.md (hunk 9 above)
Comment (cmoreno-nts) [reply README.md]: Done

File: tests/cases/helpers/case_helpers_care_platform_providers.yaml (hunk 13 above)
[the remark this answers is not in the document]
Comment (csanchez-nts) [reply tests/cases/helpers/case_helpers_care_platform_providers.yaml]: She was referring that there is a login secondary user already for provider so you didn't have to change the original provider login

File: tests/cases/helpers/case_helpers_onboarders_care_platform.yaml [hunk 15]
```diff
@@ -22,6 +22,30 @@ OnboarderCarePartnerLogin:
     - Type: sleep
       Time: 2
 
+  OnboarderCarePartnerLoginSecondaryUser:
```
Comment (csanchez-nts) [inline tests/cases/helpers/case_helpers_onboarders_care_platform.yaml:25]: Remove extra indentation

**Review by csanchez-nts — CHANGES_REQUESTED** [review]

**Review by csanchez-nts — DISMISSED** [review]

**Review by csanchez-nts — DISMISSED** [review]

**Review by dguzmanneural — APPROVED** [review]

**Review by miguel-hquez — APPROVED** [review]

**Review by csanchez-nts — APPROVED** [review]

**Pull request merged**